Automattic/jetpack
# Connection: do not display Connect Bar when in Offline mode, and make SEO userless

#### Changes proposed in this Pull Request:
﻿
This PR is a follow-up of #18837 and #19300.

It introduces the following changes:

**The connect bar main message should be translatable.**

**The SEO module now works in userless mode.**

**The Protect card should still be a bit readable when in offline mode.**

In bc422ac3f771e80a36ef47ef73d7681878411979, we wrapped the foldable card into a SettingsGroup component that already handles the display of the group when in offline mode. We consequently do not need to handle it again in the foldable card.

**Before**

<img width="1200" alt="Screenshot 2021-03-26 at 18 06 09" src="https://user-images.githubusercontent.com/426388/112670367-1e85f700-8e61-11eb-960d-ec3e1765ef78.png">


**After**

<img width="1189" alt="Screenshot 2021-03-26 at 18 06 57" src="https://user-images.githubusercontent.com/426388/112670388-22b21480-8e61-11eb-9477-1ff79f20a057.png">


**Let's not show the connect bar when in offline mode, since folks will not be able to connect their site to WordPress.com anyway.

**Before**

<img width="1075" alt="Screenshot 2021-03-26 at 18 24 02" src="https://user-images.githubusercontent.com/426388/112670409-2776c880-8e61-11eb-8659-7ddb924666c4.png">


**After**

<img width="1080" alt="Screenshot 2021-03-26 at 18 24 14" src="https://user-images.githubusercontent.com/426388/112670426-2d6ca980-8e61-11eb-808f-7f37bab0accb.png">

**Note**: a potential alternative would be to add the Offline check within the connect bar component itself. 

#### Jetpack product discussion

p1HpG7-ba8-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start with a brand new JN site running `master`.
* In the Jetpack constants page, enable Offline mode.
* If you then visit Jetpack > Settings, you should see the Userless card appear under some of the features, even though your site cannot be connected to WordPress.com.
* Switch to this branch.
* The userless connect bar prompt should disappear.
* In the Jetpack constants page, disable Offline mode and enable `JETPACK_NO_USER_TEST_MODE`.
* Disconnect from Jetpack, reconnect in userless mode.
* Go to "Settings -> Traffic -> Search engine optimization" and confirm that the module works in Userless mode (switch on/off and settings are available).
